src/main/java/com/cisco/trex/util/TRexClientUtil.java

```diff
@@ -27,7 +29,24 @@ public static TRexServerMode getMode(String host, String port) {
     parameters.put("minor", Constants.ASTF_API_VERSION_MINOR);
     RPCResponse response = null;
     try {
-      response = transport.sendCommand(buildCommand("api_sync_v2", parameters));
+      TRexCommand command = buildCommand("api_sync_v2", parameters);
+      response = transport.sendCommand(command);
+
+      String errorMessage =
+          response.getError() == null ? null : response.getError().getSpecificErr();
+      if (!StringUtils.isBlank(errorMessage) && errorMessage.contains("Version mismatch")) {
+        String regrexString = "server: '([0-9]*)\\.([0-9]*)', client: '([0-9]*)\\.([0-9]*)'";
+        Pattern pattern = Pattern.compile(regrexString);
+        Matcher matcher = pattern.matcher(errorMessage);
+        if (matcher.find()) {
+          Constants.ASTF_API_VERSION_MAJOR = Integer.parseInt(matcher.group(1));
+          Constants.ASTF_API_VERSION_MINOR = Integer.parseInt(matcher.group(2));
+          parameters.put("major", Constants.ASTF_API_VERSION_MAJOR);
+          parameters.put("minor", Constants.ASTF_API_VERSION_MINOR);
+          command = buildCommand("api_sync_v2", parameters);
+          response = transport.sendCommand(command);
```

Comment:
do NOT change constant, use local variable instead, no matter it will bring the server API version check twice here and in TRexClient and TRexAstfClient.